cmd/juju/cloud/add.go

```diff
@@ -139,8 +141,14 @@ func (c *AddCloudCommand) Run(ctxt *cmd.Context) error {
 	}
 
 	// first validate cloud input
-	if err = cloud.ValidateCloudSet([]byte(c.CloudFile)); err != nil {
-		ctxt.Warningf("%s", err.Error())
+	data, err := ioutil.ReadFile(c.CloudFile)
```

Comment:
I agree, it has the change would have tendrils however, we would be changing a shared interface.